crates/providers/src/provider.rs

```diff
@@ -280,12 +275,10 @@ impl<T: Transport + Clone + Send + Sync> TempProvider for Provider<T> {
         key: StorageKey,
         tag: Option<BlockId>,
     ) -> TransportResult<StorageValue> {
-        self.inner
-            .prepare(
-                "eth_getStorageAt",
-                (address, key, tag.unwrap_or(BlockNumberOrTag::Latest.into())),
-            )
-            .await
+        // Key is a QUANTITY according to the [spec](https://eth.wiki/json-rpc/API#eth_getstorageat).
+        // See: https://github.com/gakonst/ethers-rs/blob/b60c7fe1fea41a902e3b62dc716d45b326520fd0/ethers-providers/src/rpc/provider.rs#L674
+        let key = U256::from_be_bytes(key.into());
```

Comment:
Should this be changed in the parameter?

Reply:
I think so, the user should do that, the storageKey type is intended for the server so the off spec format is also supported

we should use spec compliance here